assignment/assignment-impl/impl/src/java/org/sakaiproject/assignment/impl/BaseAssignmentService.java

```diff
@@ -11791,9 +11791,9 @@ public String getGradeForUserInGradeBook(String userId)
 			}
 			Assignment m = getAssignment();
 			String gAssignmentName = StringUtils.trimToNull(m.getProperties().getProperty(AssignmentService.PROP_ASSIGNMENT_ASSOCIATE_GRADEBOOK_ASSIGNMENT));
-			if (gAssignmentName != null)
+			String gradebookUid = m.getContext();
+			if (gAssignmentName != null && m_gradebookService.getAssignment(gradebookUid, gAssignmentName) != null)
 			{
```

Comment:
I would refactor this a little bit. Whynot use `StringUtils.trimToEmpty()` instead of `trimtoNull()`, then you don't have to check for null in the conditional.

Also, I would move the call to `m_gradebookService.getAssignment()` outside the conditional, as this currently get's called twice which is a bit of a waste.

Something like this:

```
String gAssignmentName = StringUtils.trimToEmpty( ... );
Assignment gradebookAssignment = m_gradebookService.getAssignment( ... );
if( gradebookAssignment != null )
{
    final GradeDefinition def = m_gradebookService.getGradeDefinitionForStudentForItem( gradebookUid, gradebookAssignment.getId(), userId );
    ...
}
```